build error on my system:

```
/home/owen/src/github/mixxx/src/vinylcontrol/vinylcontrolxwax.cpp: In constructor ‘VinylControlXwax::VinylControlXwax(UserSettingsPointer, const QString&)’:
/home/owen/src/github/mixxx/src/vinylcontrol/vinylcontrolxwax.cpp:158:24: error: request for member ‘value’ in ‘sampleRate’, which is of non-class type ‘const unsigned int’
  158 |             sampleRate.value(),
      |                        ^~~~~
```

--- a/lib/xwax/timecoder.c
+++ b/lib/xwax/timecoder.c
@@ -300,7 +300,7 @@ static void init_channel(struct timecoder_channel *ch)
  */
 
 void timecoder_init(struct timecoder *tc, struct timecode_def *def,
-                    double speed, unsigned int sample_rate, bool phono)
+                    double speed, unsigned int sample_rate, bool phono, bool pitch_estimator)
 {
     assert(def != NULL);
 
@@ -321,7 +321,7 @@ void timecoder_init(struct timecoder *tc, struct timecode_def *def,
     init_channel(&tc->primary);
     init_channel(&tc->secondary);
 
-    tc->use_legacy_pitch_filter = false; /* Switch for pitch filter type */
+    tc->use_legacy_pitch_filter = pitch_estimator; /* Switch for pitch filter type */
 
     if (tc->use_legacy_pitch_filter) {
         pitch_init(&tc->pitch, tc->dt);

--- a/lib/xwax/timecoder.h
+++ b/lib/xwax/timecoder.h
@@ -90,7 +90,7 @@ struct timecode_def* timecoder_find_definition(const char *name);
 void timecoder_free_lookup(void);
 
 void timecoder_init(struct timecoder *tc, struct timecode_def *def,
-                    double speed, unsigned int sample_rate, bool phono);
+                    double speed, unsigned int sample_rate, bool phono, bool pitch_estimator);
 void timecoder_clear(struct timecoder *tc);
 
 int timecoder_monitor_init(struct timecoder *tc, int size);

--- a/src/preferences/dialog/dlgprefvinyl.cpp
+++ b/src/preferences/dialog/dlgprefvinyl.cpp
@@ -62,6 +62,15 @@ DlgPrefVinyl::DlgPrefVinyl(
         box->addItem(MIXXX_VINYL_SPEED_45);
     }
 
+    m_vcPitchEstimatorBoxes = {ComboBoxPitchEstimator1,
+            ComboBoxPitchEstimator2,
+            ComboBoxPitchEstimator3,
+            ComboBoxPitchEstimator4};
+    for (auto* box : std::as_const(m_vcPitchEstimatorBoxes)) {
+        box->addItem(MIXXX_VINYL_PITCH_FILTER_KALMAN);
+        box->addItem(MIXXX_VINYL_PITCH_FILTER_LEGACY);
+    }
+
     m_vcLeadInBoxes = {LeadinTime1, LeadinTime2, LeadinTime3, LeadinTime4};
     for (auto* box : std::as_const(m_vcLeadInBoxes)) {
         box->setSuffix(" s");
@@ -70,6 +79,7 @@ DlgPrefVinyl::DlgPrefVinyl(
     DEBUG_ASSERT(m_vcLabels.length() == kMaximumVinylControlInputs);
     DEBUG_ASSERT(m_vcTypeBoxes.length() == kMaximumVinylControlInputs);
     DEBUG_ASSERT(m_vcSpeedBoxes.length() == kMaximumVinylControlInputs);
+    DEBUG_ASSERT(m_vcPitchEstimatorBoxes.length() == kMaximumVinylControlInputs);
     DEBUG_ASSERT(m_vcLeadInBoxes.length() == kMaximumVinylControlInputs);
 
     for (int i = 0; i < kMaximumVinylControlInputs; ++i) {
@@ -198,6 +208,14 @@ void DlgPrefVinyl::slotUpdate() {
             m_vcSpeedBoxes[i]->setCurrentIndex(comboIndex);
         }
 
+        // Set pitch filter types
+        comboIndex =
+                m_vcPitchEstimatorBoxes[i]->findText(config->getValueString(
+                        ConfigKey(group, "vinylcontrol_pitch_estimator_type")));
+        if (comboIndex != -1) {
+            m_vcPitchEstimatorBoxes[i]->setCurrentIndex(comboIndex);
+        }
+
         // set lead-in time
         int leadIn = config->getValue(
                 ConfigKey(group, "vinylcontrol_lead_in_time"),
@@ -254,6 +272,8 @@ void DlgPrefVinyl::slotApply() {
                 ConfigValue(m_vcTypeBoxes[i]->currentText()));
         config->set(ConfigKey(group, "vinylcontrol_speed_type"),
                 ConfigValue(m_vcSpeedBoxes[i]->currentText()));
+        config->set(ConfigKey(group, "vinylcontrol_pitch_estimator_type"),
+                ConfigValue(m_vcPitchEstimatorBoxes[i]->currentText()));
         verifyAndSaveLeadInTime(i, group);
 
         m_signalWidgets[i]->setVinylActive(m_pVCManager->vinylInputConnected(i));
@@ -306,12 +326,14 @@ void DlgPrefVinyl::setDeckWidgetsVisible(int deck, bool visible) {
         m_vcLabels[deck]->show();
         m_vcTypeBoxes[deck]->show();
         m_vcSpeedBoxes[deck]->show();
+        m_vcPitchEstimatorBoxes[deck]->show();
         m_vcLeadInBoxes[deck]->show();
         m_signalWidgets[deck]->show();
     } else {
         m_vcLabels[deck]->hide();
         m_vcTypeBoxes[deck]->hide();
         m_vcSpeedBoxes[deck]->hide();
+        m_vcPitchEstimatorBoxes[deck]->hide();
         m_vcLeadInBoxes[deck]->hide();
         m_signalWidgets[deck]->hide();
     }

--- a/src/preferences/dialog/dlgprefvinyl.h
+++ b/src/preferences/dialog/dlgprefvinyl.h
@@ -46,6 +46,7 @@ class DlgPrefVinyl : public DlgPreferencePage, Ui::DlgPrefVinylDlg  {
     QList<QLabel*> m_vcLabels;
     QList<QComboBox*> m_vcTypeBoxes;
     QList<QComboBox*> m_vcSpeedBoxes;
+    QList<QComboBox*> m_vcPitchEstimatorBoxes;
     QList<QSpinBox*> m_vcLeadInBoxes;
     QList<VinylControlSignalWidget*> m_signalWidgets;
 

--- a/src/preferences/dialog/dlgprefvinyldlg.ui
+++ b/src/preferences/dialog/dlgprefvinyldlg.ui
@@ -187,6 +187,30 @@
        </widget>
       </item>
 
+      <item row="0" column="4">
+       <widget class="QLabel" name="PitchEstimatorLabel">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="leftMargin">
+         <number>3</number>
+        </property>
+        <property name="text">
+         <string>Pitch estimator</string>
+        </property>
+        <property name="wordWrap">
+         <bool>false</bool>
+        </property>
+       </widget>
+      </item>
+
+
       <item row="1" column="0">
        <widget class="QLabel" name="VinylLabel1">
         <property name="enabled">
@@ -542,6 +566,123 @@
         </property>
        </widget>
       </item>
+
+      <item row="1" column="4">
+       <widget class="QComboBox" name="ComboBoxPitchEstimator1">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="font">
+         <font/>
+        </property>
+        <property name="minimumSize">
+         <size>
+          <width>100</width>
+          <height>0</height>
+         </size>
+        </property>
+        <property name="maximumSize">
+         <size>
+          <width>200</width>
+          <height>16777215</height>
+         </size>
+        </property>
+       </widget>
+      </item>
+
+      <item row="2" column="4">
+       <widget class="QComboBox" name="ComboBoxPitchEstimator2">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="font">
+         <font/>
+        </property>
+        <property name="minimumSize">
+         <size>
+          <width>100</width>
+          <height>0</height>
+         </size>
+        </property>
+        <property name="maximumSize">
+         <size>
+          <width>200</width>
+          <height>16777215</height>
+         </size>
+        </property>
+       </widget>
+      </item>
+
+      <item row="3" column="4">
+       <widget class="QComboBox" name="ComboBoxPitchEstimator3">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="font">
+         <font/>
+        </property>
+        <property name="minimumSize">
+         <size>
+          <width>100</width>
+          <height>0</height>
+         </size>
+        </property>
+        <property name="maximumSize">
+         <size>
+          <width>200</width>
+          <height>16777215</height>
+         </size>
+        </property>
+       </widget>
+      </item>
+
+      <item row="4" column="4">
+       <widget class="QComboBox" name="ComboBoxPitchEstimator4">
+        <property name="enabled">
+         <bool>true</bool>
+        </property>
+        <property name="sizePolicy">
+         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
+          <horstretch>0</horstretch>
+          <verstretch>0</verstretch>
+         </sizepolicy>
+        </property>
+        <property name="font">
+         <font/>
+        </property>
+        <property name="minimumSize">
+         <size>
+          <width>100</width>
+          <height>0</height>
+         </size>
+        </property>
+        <property name="maximumSize">
+         <size>
+          <width>200</width>
+          <height>16777215</height>
+         </size>
+        </property>
+       </widget>
+      </item>
+
 
       <item row="1" column="4" rowspan="4">
        <spacer name="horizontalSpacer_3">

--- a/src/test/co_dumps/co_dump_inital.csv
+++ b/src/test/co_dumps/co_dump_inital.csv
@@ -5028,6 +5028,7 @@
 [Channel2],hotcue_11_activate_preview,0
 [Sampler1],hotcue_27_enabled,0
 [Channel3],vinylcontrol_speed_type,33.3
+[Channel3],vinylcontrol_pitch_estimator_type,"Kalman filter (accurate)"
 [Sampler3],beatloop_0.0625_enabled,0
 [Channel3],hotcue_19_set,0
 [EqualizerRack1_[Channel2]],enabled,1
@@ -5038,6 +5039,7 @@
 [Channel1],playposition_up,0
 [EqualizerRack1_[Channel2]_Effect1],parameter6_set_default,0
 [Channel4],vinylcontrol_speed_type,33.3
+[Channel4],vinylcontrol_pitch_estimator_type,"Kalman filter (accurate)"
 [PreviewDeck1],hotcue_4_enabled,0
 [Channel4],hotcue_19_set,0
 [EqualizerRack1_[Channel4]_Effect1],parameter15_set_zero,0
@@ -5200,6 +5202,7 @@
 [Channel2],hotcue_20_gotoandstop,0
 [EffectRack1_EffectUnit3_Effect1],parameter12_set_default,0
 [Channel1],vinylcontrol_speed_type,33.3
+[Channel1],vinylcontrol_pitch_estimator_type,"Kalman filter (accurate)"
 [QuickEffectRack1_[Channel4]_Effect1],parameter6_up,0
 [EqualizerRack1_[Channel1]_Effect1],parameter6_down,0
 [PreviewDeck1],hotcue_8_goto,0
@@ -5211,6 +5214,7 @@
 [Channel3],hotcue_20_gotoandstop,0
 [QuickEffectRack1_[Channel3]_Effect1],parameter3_set_minus_one,0
 [Channel2],vinylcontrol_speed_type,33.3
+[Channel2],vinylcontrol_pitch_estimator_type,"Kalman filter (accurate)"
 [PreviewDeck1],hotcue_27_activate_preview,0
 [EffectRack1_EffectUnit2],loaded,1
 [EqualizerRack1_[Channel4]_Effect1],parameter8_link_inverse,0

--- a/src/vinylcontrol/defs_vinylcontrol.h
+++ b/src/vinylcontrol/defs_vinylcontrol.h
@@ -44,6 +44,9 @@ constexpr int VINYL_STATUS_ERROR = 3;
 #define MIXXX_VINYL_SPEED_33 "33.3 RPM"
 #define MIXXX_VINYL_SPEED_45 "45 RPM"
 
+#define MIXXX_VINYL_PITCH_FILTER_LEGACY "Alpha-Beta filter (fast)"
+#define MIXXX_VINYL_PITCH_FILTER_KALMAN "Kalman filter (accurate)"
+
 #define MIXXX_VINYL_SPEED_33_NUM (100.0 / 3.0)
 #define MIXXX_VINYL_SPEED_45_NUM 45.0
 

--- a/src/vinylcontrol/vinylcontrolxwax.cpp
+++ b/src/vinylcontrol/vinylcontrolxwax.cpp
@@ -71,6 +71,8 @@ VinylControlXwax::VinylControlXwax(UserSettingsPointer pConfig, const QString& g
         ConfigKey(group,"vinylcontrol_vinyl_type"));
     QString strVinylSpeed = m_pConfig->getValueString(
         ConfigKey(group,"vinylcontrol_speed_type"));
+    QString strPitchEstimator = m_pConfig->getValueString(
+            ConfigKey(group, "vinylcontrol_pitch_estimator_type"));
 
     // libxwax indexes by C-strings so we pass libxwax string literals so we
     // don't have to deal with freeing the strings later
@@ -148,7 +150,14 @@ VinylControlXwax::VinylControlXwax(UserSettingsPointer pConfig, const QString& g
     // do this once across the VinylControlXwax instances.
     s_xwaxLUTMutex.lock();
 
-    timecoder_init(&timecoder, tc_def, speed, sampleRate, /* phono */ false);
+    const bool use_legacy_pitch_filter = strPitchEstimator == MIXXX_VINYL_PITCH_FILTER_LEGACY;
+
+    timecoder_init(&timecoder,
+            tc_def,
+            speed,
+            sampleRate,
+            /* phono */ false,
+            use_legacy_pitch_filter);
     timecoder_monitor_init(&timecoder, MIXXX_VINYL_SCOPE_SIZE);
     //Note that timecoder_init will not double-malloc the LUTs, and after this we are guaranteed
     //that the LUT has been generated unless we ran out of memory.
@@ -513,7 +522,6 @@ void VinylControlXwax::analyzeSamples(CSAMPLE* pSamples, size_t nFrames) {
         }
 
         if (uiUpdateTime(filePosition)) {
-
             // Don't show extremely high or low speeds in the UI.
             if (reportedPlayButton && !scratching->toBool() &&
                     dVinylPitch < 1.9 && dVinylPitch > 0.2) {

